spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -603,11 +603,14 @@ public void testUpdateWithSelfSubquery() {
         ImmutableList.of(row(1, "hr"), row(2, "x")),
         sql("SELECT * FROM %s ORDER BY id", tableName));
 
-    sql("UPDATE %s SET dep = 'y' WHERE " +
-        "id = (SELECT count(*) FROM (SELECT DISTINCT id FROM %s) AS t)", tableName, tableName);
-    assertEquals("Should have expected rows",
-        ImmutableList.of(row(1, "hr"), row(2, "y")),
-        sql("SELECT * FROM %s ORDER BY id", tableName));
+    // TODO: Spark does not support AQE and DPP with aggregates at the moment
```

Comment:
This fails if AQE is enabled?

Reply:
Unfortunately, yes. It contains a distinct aggregation.